### PR TITLE
Fix right click context menu on ag-grid tables

### LINKED ISSUE #5153: Bug - TO right click menu on servers tab is sometimes cut off if you have reached end of list
## I am submitting a...

-  bug report

-  Traffic Portal Version: 4.1.0-10702.c5c3723f.el7

## Current behavior:
When reaching the end of a server list, and right clicking on the host for quick options, the popup menu gets cut off 

## Expected behavior:
menu should display above where there is room to display all options

## Minimal reproduction of the problem with instructions:
image
if you notice in this image, i have both the table scrolled to the bottom and the outer page. instead of cutting off the pop up menu, it woudl be nice for it to move to the top instead of cuttign off as there is no way around it.
Test in Latest Firefox and Safari

### PR DESCRIPTION
**Pull request by shamrickus**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5153 

The right click menu had it's position set to the _absolute_ position of the mouse on the page instead of the _relative_ position.

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
Ensure on the servers/jobs table, that after scrolling the page _vertically_ - the right click context menu appears where the mouse is.

## If this is a bug fix, what versions of Traffic Control are affected?

- master (5dcef0a)
- 4.1.0 (de04978)

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: CHANGELOG.md
Comment (mitchell852) [inline CHANGELOG.md:66]: what happened here? did you just put them in order? OCD? :)

Comment (shamrickus) [reply CHANGELOG.md:66]: Yeah, just sorted by issue number. Realized there were more. I'm gonna back it out and just have my change since at this point a clean-up of the changelog would probably be more appropriate.

Comment (mitchell852) [reply CHANGELOG.md:66]: yeah maybe @ocket8888 (the release manager) can clean it up or work with you to clean it up as the last commit prior to cutting the 5.x branch.

**Review by mitchell852 — COMMENTED** [review]
What about when it rolls off the bottom of the screen like such?

image

not sure off the top of my head how to make the context menu go up rather than down when no room at bottom...

**Comment by shamrickus** [comment]
The menu should now detect if it's cut off on the bottom and/or the right, and if so it will position it on the opposite side.

**Review by mitchell852 — COMMENTED** [review]
this is great! one small request. think in table.servers.tpl.html you can change

```
<a ng-href="#!/servers/{{server.id}}" target="_blank">Open in New Tab</a>
```
to
```
<a ng-href="#!/servers/{{server.id}}" target="_blank">Open {{server.hostName}} in New Tab</a>
```
so there is no confusion about which row the context menu is associated with...

**Review by mitchell852 — APPROVED** [review]
LGTM! I'll merge once the GHAs all pass.

**Pull request merged**
